File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        set<int> s1,s2;
        for(int i=0;i<nums1.size();i++){
            s1.insert(nums1[i]);
        }
        for(int i=0;i<nums2.size();i++){
            s2.insert(nums2[i]);
        }
        vector<vector<int>> ans(2);
        for(auto x: s1){
            if(s2.find(x)==s2.end()){
                ans[0].push_back(x);
            }
        }
        for(auto x: s2){
            if(s1.find(x)==s1.end()){
                ans[1].push_back(x);
            }
        }
        return ans;
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> s1 = new HashSet<>();
        Set<Integer> s2 = new HashSet<>();
        for(int x: nums1){
            s1.add(x);
        }
        for(int x: nums2){
            s2.add(x);
        }
        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>());
        ans.add(new ArrayList<>());
        for(int x: s1){
            if(!s2.contains(x)){
                ans.get(0).add(x);
            }
        }
        for(int x: s2){
            if(!s1.contains(x)){
                ans.get(1).add(x);
            }
        }
        return ans;
    }
}